proposed/Launcher-Service-Documentation.md

```diff
@@ -0,0 +1,186 @@
+- Feature Name: Launcher service documentation
+- Type: New Product
+- Related components: safe_launcher_core (accompanying RFC)
+- Start Date: 05-October-2015
+
+# Summary
+
+This is an accompanying RFC to the parent `Launcher-as-a-service` RFC and defines JSONs for Launcher-App RPC. This RFC must be read and is useful only in conjunction with the parent RFC.
+
+# Conventions
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+# Detailed design
+
+## Globals
+- RSA-Key-Exchange, app to Launcher
+```
+{
+    "rsa_key_exchange_request": {
+        "launcher_string": UTF-8 String,        // This shall be the one supplied by Launcher
+        "nonce": [ uint8 ... ],                 // sodiumoxide::crypto::box_::Nonce,
+        "public_encryption_key": [ uint8 ... ]  // sodiumoxide::crypto::box_::PublicKey from
+                                                // <App-Asymm-Keys>
+    }
+}
+```
+- RSA-Key-Exchange, Launcher to app
+```
+{
+    "rsa_key_exchange_response": {
+        "cipher_text": [ uint8 ... ] // encrypted symmetric keys
+    }
+}
+```
+- Version Negotiation
+```
+{
+    "version": x.y // where x.y could be 2.10 etc
+}
+```
+- Errors
+```
+{
+    "error": {
+        "error_code": Integer, // This shall be whatever Into trait provides in various modules
+        "error_description": String
+    }
+}
+```
+
+## Module-Specific
+- Modules
+```
+"NFS"
+"DNS"
+```
+
+### NFS
+- Actions
+```
+"create-dir"
+"delete-dir"
+"create-file"
+"delete-file"
+```
+
+- Create Directory
```

Comment:
Nice start for an initial implementation to the RPCs. Might be worth considering some alternatives to the data section formats for the future too. say for example:
```
{
  endpoint: "safe-api/v1.4/nfs/create-dir"
  data: {
    is_shared: false,
    path: "/a/b/c"
    is_private: true
    ...
}
```